controller/appcontroller.go

```diff
@@ -567,6 +567,7 @@ func (ctrl *ApplicationController) getResourceTree(a *appv1.Application, managed
 			managedResourcesKeys = append(managedResourcesKeys, kube.GetResourceKey(live))
 		}
 	}
+	nodesMap := make(map[kube.ResourceKey]appv1.ResourceNode)
 	err = ctrl.stateCache.IterateHierarchyV2(a.Spec.Destination.Server, managedResourcesKeys, func(child appv1.ResourceNode, appName string) bool {
```

Comment:
I think this problem is caused by the IterateHierarchyV2 implementation. It is unexpected to have the callback happen twice for the same node. The fix would need to be implemented in https://github.com/argoproj/gitops-engine. I think @andrii-korotkov-verkada can helm you figure it out, but you should be able to write a unit tests in that repo and test it when a resource has 2 owner references.

Reply:
Thank you for your analysis. I will take a look at the IterateHierarchyV2 implementation and see if I can address the issue in the https://github.com/argoproj/gitops-engine repository. I also appreciate your suggestion to involve @andrii-korotkov-verkada. I will reach out to him/her for assistance. Additionally, I will work on writing unit tests as you suggested and test when a resource has two owner references. Thank you again for your valuable input.

Reply:
What's the use case for having multiple owner references? My understanding is that this breaks a tree-like structure of the resources and not sure what's the best way to support this. We can pick one of parents, though I thought that's what should be happening based on visited flagging. Do you see a whole sub-tree being added or only the top-most entry?